datafusion/substrait/tests/roundtrip_logical_plan.rs

```diff
@@ -262,7 +262,79 @@ mod tests {
 
     #[tokio::test]
     async fn qualified_catalog_schema_table_reference() -> Result<()> {
-        roundtrip("SELECT * FROM datafusion.public.data;").await
+        roundtrip("SELECT a,b,c,d,e FROM datafusion.public.data;").await
+    }
+
+    #[tokio::test]
+    async fn read_all_types() -> Result<()> {
+        let mut ctx = create_all_type_context().await?;
+        let df = ctx
+            .sql(
+                "select * from data where 
+                a = TRUE AND
+                b = 0 AND
+                c = 0 AND
+                d = 0 AND
+                e = 0 AND
+                f = 0 AND
+                g = 0 AND
+                h = 0 AND
+                i = 0;",
+            )
+            .await?;
+        println!("{:?}", df.clone().collect().await.unwrap());
+        let plan = df.into_optimized_plan()?;
+        let proto = to_substrait_plan(&plan)?;
+        let plan2 = from_substrait_plan(&mut ctx, &proto).await?;
+        let plan2 = ctx.state().optimize(&plan2)?;
+
+        println!("{plan:#?}");
+        println!("{plan2:#?}");
+
+        let plan1str = format!("{plan:?}");
+        let plan2str = format!("{plan2:?}");
+        assert_eq!(plan1str, plan2str);
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn integer_type_literal() -> Result<()> {
+        roundtrip_all_types(
+            "select * from data where 
+            a = TRUE AND
+            b = 0 AND
+            c = 0 AND
+            d = 0 AND
+            e = 0 AND
+            f = 0 AND
+            g = 0 AND
+            h = 0 AND
+            i = 0;",
```

Comment:
I wonder if there is any reason for not testing  `j` and `k` (f32 and f64) as well?